api/src/services/survey-service.ts

```diff
@@ -67,11 +100,114 @@ export class SurveyService extends DBService {
       throw new HTTP400('Failed to get species data');
     }
 
+    //parse result from DB into focal and ancillary species
+    const parsedSpeciesIds = new ParsedSpeciesIds(result);
+
     const taxonomyService = new TaxonomyService();
 
-    const species = await taxonomyService.getSpeciesFromIds(result);
+    const focal_species_result = await taxonomyService.getSpeciesFromIds(parsedSpeciesIds.focal_species);
+    const ancillary_species_result = await taxonomyService.getSpeciesFromIds(parsedSpeciesIds.ancillary_species);
+
+    const focal_species = new GetFocalSpeciesData(focal_species_result);
+    const ancillary_species = new GetAncillarySpeciesData(ancillary_species_result);
+
+    return {
+      focal_species: focal_species.focal_species,
+      focal_species_names: focal_species.focal_species_names,
+      ancillary_species: ancillary_species.ancillary_species,
+      ancillary_species_names: ancillary_species.ancillary_species_names
+    };
+  }
+
+  async getPermitData(surveyId: number): Promise<GetPermitData> {
+    const sqlStatement = SQL`
+      SELECT
+        number, type
+      FROM
+        permit
+      WHERE
+        survey_id = ${surveyId};
+      `;
+
+    const response = await this.connection.query(sqlStatement.text, sqlStatement.values);
+
+    const result = (response && response.rows[0]) || null;
+
+    if (!result) {
+      throw new HTTP400('Failed to get permit data');
+    }
+
+    return new GetPermitData(result);
+  }
+
+  async getSurveyPurposeAndMethodology(surveyId: number): Promise<object> {
+    const sqlStatement = queries.survey.getSurveyPurposeAndMethodologyForUpdateSQL(surveyId);
+
+    if (!sqlStatement) {
+      throw new HTTP400('Failed to build SQL get statement');
+    }
+
+    const response = await this.connection.query(sqlStatement.text, sqlStatement.values);
+
+    if (!response || !response?.rows?.[0]) {
```

Comment:
Should the error check convention be continued here? Similar to above and below.